src/listpack.c

```diff
@@ -793,40 +831,54 @@ unsigned char *lpInsert(unsigned char *lp, unsigned char *elestr, unsigned char
      * address again after a reallocation. */
     unsigned long poff = p-lp;
 
-    int enctype;
-    if (elestr) {
-        /* Calling lpEncodeGetType() results into the encoded version of the
-        * element to be stored into 'intenc' in case it is representable as
-        * an integer: in that case, the function returns LP_ENCODING_INT.
-        * Otherwise if LP_ENCODING_STR is returned, we'll have to call
-        * lpEncodeString() to actually write the encoded string on place later.
-        *
-        * Whatever the returned encoding is, 'enclen' is populated with the
-        * length of the encoded element. */
-        enctype = lpEncodeGetType(elestr,size,intenc,&enclen);
-        if (enctype == LP_ENCODING_INT) eleint = intenc;
-    } else if (eleint) {
-        enctype = LP_ENCODING_INT;
-        enclen = size; /* 'size' is the length of the encoded integer element. */
-    } else {
-        enctype = -1;
-        enclen = 0;
+    unsigned char *rep = p; /* Replaced element in case where == LP_REPLACE */
+    for (unsigned long i = 0; i < len; i++) {
+        if (entries) {
+            listpackEntry *e = &entries[i];
+            if (e->sval) {
+                /* Calling lpEncodeGetType() results into the encoded version of the
+                * element to be stored into 'intenc' in case it is representable as
+                * an integer: in that case, the function returns LP_ENCODING_INT.
+                * Otherwise if LP_ENCODING_STR is returned, we'll have to call
+                * lpEncodeString() to actually write the encoded string on place later.
+                *
+                * Whatever the returned encoding is, 'enclen' is populated with the
+                * length of the encoded element. */
+                e->enctype = lpEncodeGetType(e->sval, e->slen, e->intenc, &e->enclen);
+            } else {
+                e->enctype = LP_ENCODING_INT;
+                lpEncodeIntegerGetType(e->lval, e->intenc, &e->enclen);
+            }
+            addedlen += e->enclen;
+            e->backlen_size = lpEncodeBacklen(e->backlen, e->enclen);
+            addedlen += e->backlen_size;
+            n_added++;
+        }
+
+        if (where == LP_REPLACE) {
+            assert(rep[0] != LP_EOF);
```

Comment:
@sundb Another thing, if lp has 1 entry and you call LP_REPLACE with 2 entries, it will trigger the assert. Let me know if you think we should handle it silently. e.g. just delete the existing one and add new ones.